### ISSUE TITLE
Raise a warning on adding duplicated resource

### ISSUE THREAD

**Issue by asvetlov** [aio-libs/aiohttp#3023]
Now, `add_route` merges consequenced `add_get(path, handler)` and `add_post(path, handler)` calls but doesn't do it if a resource for other path is pushed in between calls.
Also, it doesn't merge the same resource but under different names.

Let's not change behavior but raise a user warning for such cases, most likely it is the result of silly mistake/misunderstanding.

**Comment by pposca** [comment]
What is the reason for merging only consequenced calls? I mean, if you have to detect non-consecutive calls or the same resource under different names to send a warning, why not just merge all cases?

**Comment by asvetlov** [comment]
Good question!

Consider the following:
```
[get('/{arg}', h1),
 post(r'/{num:\d+}', h2),
 post('/{arg}', h3)]
```
Looks weird (and sure it is).
But the example demonstrates the real problem.
By merging `/{arg}` handlers `/{num:\d+}` will be never called.

**Comment by pposca** [comment]
Of course, the order is important. I didn't realize it. Now, I understand it. I will have a look at it. Thanks.

**Comment by asvetlov** [comment]
Cool!
